crates/oxc_minifier/src/ast_passes/convert_to_dotted_properties.rs

```diff
@@ -0,0 +1,277 @@
+    ) {
+        self.convert_computed_member_expression(expr, ctx);
+    }
+}
+
+impl<'a> ConvertToDottedProperties {
+    pub fn new() -> Self {
+        Self { changed: false }
+    }
+
+    fn convert_computed_member_expression(
+        &mut self,
+        expr: &mut MemberExpression<'a>,
+        ctx: &mut TraverseCtx<'a>,
+    ) {
+        if let MemberExpression::ComputedMemberExpression(computed_expr) = expr {
+            if let Expression::StringLiteral(key) = &computed_expr.expression {
+                if is_identifier_name(&key.value) {
+                    let property = ctx.ast.identifier_name(key.span, key.value.clone());
+                    *expr = ctx.ast.member_expression_static(
+                        computed_expr.span(),
+                        ctx.ast.move_expression(&mut computed_expr.object),
+                        property,
+                        computed_expr.optional,
+                    );
+                    self.changed = true;
+                }
+            }
+        }
+    }
+}
+
+/// Port from <https://github.com/google/closure-compiler/blob/v20240609/test/com/google/javascript/jscomp/ConvertToDottedPropertiesTest.java>
+#[cfg(test)]
+mod test {
+    use oxc_allocator::Allocator;
+
+    use crate::tester;
+
+    fn test(source_text: &str, expected: &str) {
+        let allocator = Allocator::default();
+        let mut pass = super::ConvertToDottedProperties::new();
+        tester::test(&allocator, source_text, expected, &mut pass);
+    }
+
+    fn test_same(source_text: &str) {
+        test(source_text, source_text);
+    }
+
+    #[test]
+    fn test_convert() {
+        test("a['p']", "a.p");
+        test("a['_p_']", "a._p_");
+        test("a['_']", "a._");
+        test("a['$']", "a.$");
+        test("a.b.c['p']", "a.b.c.p");
+        test("a.b['c'].p", "a.b.c.p");
+        test("a['p']();", "a.p();");
+        test("a()['p']", "a().p");
+        // ASCII in Unicode is safe.
+        test("a['\\u0041A']", "a.AA");
+    }
+
+    #[test]
+    fn test_do_not_convert() {
+        test_same("a[0]");
+        test_same("a['']");
+        test_same("a[' ']");
+        test_same("a[',']");
+        test_same("a[';']");
+        test_same("a[':']");
+        test_same("a['.']");
+        test_same("a['0']");
+        test_same("a['p ']");
+        test_same("a['p' + '']");
+        test_same("a[p]");
+        test_same("a[P]");
+        test_same("a[$]");
+        test_same("a[p()]");
+        // test_same("a['default']");
+        // Ignorable control characters are ok in Java identifiers, but not in JS.
+        test_same("a['A\\u0004']");
+        // upper case lower half of o from phonetic extensions set.
+        // valid in Safari, not in Firefox, IE.
+        // test_same("a['\\u1d17A']");
+        // Latin capital N with tilde - nice if we handled it, but for now let's
+        // only allow simple Latin (aka ASCII) to be converted.
+        // test_same("a['\\u00d1StuffAfter']");
```

Comment:
`\u1d17A` is invalid for ES5 (ES5 uses Unicode 3.0 and that character was introduced in Unicode 3.1) and valid for ES2015 (ES2015 uses Unicode 5.1).
`\u00d1` is valid for ES3+.